### PR TITLE
Make test-e2e to run test with local operator

### PR DESCRIPTION
**Pull request by baijum**
Use `--up-local` option to run the test

### CODE REVIEW

File: make/test.mk [hunk 1]
```diff
@@ -32,33 +32,20 @@ get-test-namespace: ./out/test-namespace
...
 	$(info Running E2E test: $@)
-	$(Q)go test ./test/e2e/... \
-		-parallel=1 \
-		${V_FLAG} \
-		-root=$(PWD) \
-		-kubeconfig=$(HOME)/.kube/config \
-		-globalMan ./deploy/test/global-manifests.yaml \
-		-namespacedMan ./deploy/test/namespace-manifests.yaml \
-		-singleNamespace
+	$(eval DEPLOYED_NAMESPACE := $(TEST_NAMESPACE))
```
Comment (DhritiShikhar) [inline make/test.mk]: @baijum what is the need to store the `TEST_NAMESPACE` value as `DEPLOYED_NAMESPACE` and then use it in the next line?

File: make/test.mk [hunk 2]
```diff
@@ -32,33 +32,23 @@ get-test-namespace: ./out/test-namespace
...
-## TODO: TBD
 e2e-cleanup: get-test-namespace
-	$(Q)-oc login -u system:admin
-	$(Q)-oc delete -f ./deploy/crds/devconsole_v1alpha1_component_crd.yaml
-	$(Q)-oc delete -f ./deploy/crds/devconsole_v1alpha1_gitsource_crd.yaml
-	$(Q)-oc delete -f ./deploy/service_account.yaml --namespace $(TEST_NAMESPACE)
-	$(Q)-oc delete -f ./deploy/role.yaml --namespace $(TEST_NAMESPACE)
-	$(Q)-oc delete -f ./deploy/test/role_binding_test.yaml --namespace $(TEST_NAMESPACE)
-	$(Q)-oc delete -f ./deploy/test/operator_test.yaml --namespace $(TEST_NAMESPACE)
+	$(Q)-oc delete project $(TEST_NAMESPACE)  --wait
```
Comment (DhritiShikhar) [inline make/test.mk]: Is there a need to add `--timeout` here as well?

File: make/test.mk (hunk 1 above)
Comment (baijum) [reply make/test.mk]: @DhritiShikhar That `DEPLOYED_NAMESPACE` environment variable is later used in the test. The test was already written before. If you see my change in `test/e2e/component_test.go` in this PR, you can see the location where it is used.  The environment variable is used to identify the namespaces. Later inside the test, a wait is added for the operator deployment to take place.

File: make/test.mk (hunk 2 above)
Comment (baijum) [reply make/test.mk]: You mean the `--wait` argument? That is required to ensure the project (namespace) has been deleted.

File: make/test.mk (hunk 1 above)
Comment (DhritiShikhar) [reply make/test.mk]: addressed here 
efa601c

File: make/test.mk (hunk 2 above)
Comment (DhritiShikhar) [reply make/test.mk]: addressed here 
4aaa60c

**Pull request merged**